application/src/main/java/org/opentripplanner/gtfs/graphbuilder/GtfsModule.java

```diff
@@ -388,9 +388,15 @@ private GtfsMutableRelationalDao loadBundle(GtfsBundle gtfsBundle) throws IOExce
       fareProduct.getId().setAgencyId(reader.getDefaultAgencyId());
     }
     for (var transferRule : store.getAllEntitiesForType(FareTransferRule.class)) {
-      transferRule.getFareProductId().setAgencyId(reader.getDefaultAgencyId());
-      transferRule.getFromLegGroupId().setAgencyId(reader.getDefaultAgencyId());
-      transferRule.getToLegGroupId().setAgencyId(reader.getDefaultAgencyId());
+      if (transferRule.getFareProductId() != null) {
+        transferRule.getFareProductId().setAgencyId(reader.getDefaultAgencyId());
```

Comment:
Not necessarily related: does it make sense to extract a variable for `reader.getDefaultAgencyId()` (that's going to touch many more lines, though)?

Reply:
Yes, it would make sense. I will send a follow up where I do a bit of refactoring of GtfsModule but not now because @t2gran is currently refactoring it as well.